### PR TITLE
Infra upgrade for v5.x.x

### PR DESCRIPTION
**Pull request by mihails-strasuns-sociomantic**
- Upgrade to HEAD of ocean v4.x.x
- Start testing with dmd-transitional 2.071.2.s2

### CODE REVIEW

**Comment by mihails-strasuns-sociomantic** [comment]
Anyone? :(

**Comment by gavin-norman-sociomantic** [comment]
Could you put the subsum output in the commit message?

**Comment by gavin-norman-sociomantic** [comment]
Also, does this imply an update to the dependencies listed in the README?

**Review by david-eckardt-sociomantic — APPROVED** [review]

File: src/swarm/node/model/NeoNode.d [hunk 1]
```diff
@@ -870,7 +870,8 @@ public class NodeBase ( ConnHandler : ISwarmConnectionHandler ) : INodeBase
     ***************************************************************************/
 
     private void handleUpdateCredentials ( cstring args,
-        void delegate ( cstring response ) send_response )
+        void delegate ( cstring response ) send_response,
+        void delegate ( ref mstring response ) wait_reply )
```
Comment (gavin-norman-sociomantic) [inline src/swarm/node/model/NeoNode.d:874]: @nemanja-boric-sociomantic do we need to change anything more than this here?

File: test/neo/client/Client.d [hunk 2]
```diff
@@ -31,7 +31,7 @@ public class Client
 
     ***************************************************************************/
 
-    private class Neo
+    public class Neo
```
Comment (gavin-norman-sociomantic) [inline test/neo/client/Client.d:34]: Huh. This is not intended to be public. Why is this change required?

**Review by gavin-norman-sociomantic — COMMENTED** [review]
Hmm... the protection attribute changes are interesting...

File: .travis.yml [hunk 3]
```diff
@@ -23,6 +23,8 @@ env:
         - DMD=1.080.0 DIST=xenial F=devel
         - DMD=2.070.2.s12 DIST=xenial F=production
         - DMD=2.070.2.s12 DIST=xenial F=devel
+        - DMD=2.071.2.s2 DIST=xenial F=production
+        - DMD=2.071.2.s2 DIST=xenial F=devel
```
Comment (nemanja-boric-sociomantic) [inline .travis.yml]: Can we use wildcard here?

File: src/swarm/node/model/NeoNode.d (hunk 1 above)
Comment (nemanja-boric-sociomantic) [reply src/swarm/node/model/NeoNode.d:874]: No, I think that's it. It just enables you to call `wait_reply` from the handler delegate. No more work is needed.

File: test/neo/client/Client.d (hunk 2 above)
Comment (mihails-strasuns-sociomantic) [reply test/neo/client/Client.d:34]: It is used from `test/neo/main.d`

Comment (mihails-strasuns-sociomantic) [reply test/neo/client/Client.d:34]: also `test/client_auth/main.d`

**Comment by mihails-strasuns-sociomantic** [comment]
> Also, does this imply an update to the dependencies listed in the README?

It depends on HEAD branch state so I am unsure what to put into README. Makes more sense to update it when cleaning up to use actual release eventually.

File: test/neo/client/Client.d (hunk 2 above)
Comment (gavin-norman-sociomantic) [reply test/neo/client/Client.d:34]: Ah yes, of course. It's used for the various aliases it contains. I think this should be a case for a private ctor.

**Comment by gavin-norman-sociomantic** [comment]
>Makes more sense to update it when cleaning up to use actual release eventually.

:+1:

File: test/neo/client/Client.d (hunk 2 above)
Comment (gavin-norman-sociomantic) [reply test/neo/client/Client.d:34]: `private class Blocking` will need the same treatment.

Comment (gavin-norman-sociomantic) [reply test/neo/client/Client.d:34]: Actually, no, sorry, it only contains methods, which can be accessed via the instance.

Comment (gavin-norman-sociomantic) [reply test/neo/client/Client.d:34]: Actually, no, sorry 2: `Blocking` mixes is `TaskBlockingCore`, which contains types which will need to be accessible from outside.

Comment (mihails-strasuns-sociomantic) [reply test/neo/client/Client.d:34]: Whatever. Can you do any change you find necessary as a separate PR? I am only fixing compiler errors for existing code :P

**Comment by mihails-strasuns-sociomantic** [comment]
Updated matrix with wildcard

File: test/neo/client/Client.d (hunk 2 above)
Comment (gavin-norman-sociomantic) [reply test/neo/client/Client.d:34]: Sure. Can you at least change the ctor of `Neo` to be private in the same commit where you make the class public?

Comment (mihails-strasuns-sociomantic) [reply test/neo/client/Client.d:34]: It is already private because it is mixed in by `ClientCore`

Comment (gavin-norman-sociomantic) [reply test/neo/client/Client.d:34]: Ah excellent. Thanks for checking.

**Review by gavin-norman-sociomantic — APPROVED** [review]
